actfw_core/capture.py

```diff
@@ -61,6 +71,9 @@ def __init__(
             try to capture one of the fallback_formats and convert it to expected_format.
 
         """
+        if fallback_formats is None:
+            fallback_formats = [V4L2_PIX_FMT.YUYV, V4L2_PIX_FMT.MJPEG]
+
```

Comment:
Why do you allow `None` as `fallback_formats`?

Reply:
Due to an error of flake8:

```
/home/keno/src/github.com/Idein/actfw-core/actfw_core/capture.py:50:48: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
```

FYI, this pattern is forbidden in python (because there isn't "mutability"/"ownership" in python nor a sufficiently smart static analyzer, AFAIK). B006 includes the following pattern and we shouldn't use `noqa` carelessly.

```
$  python
Python 3.9.1 (default, Apr  4 2021, 01:39:14)
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def f(xs = [0, 1]):
...     return xs
...
>>> f().append(100)
>>> f()
[0, 1, 100]
```

Allowing `None` gives us a chance to check "the argument is explicitly passed or not".

Or, do you mean the following is preferable?

```
diff --git a/actfw_core/capture.py b/actfw_core/capture.py
index fe52f2a..60f8cf5 100644
--- a/actfw_core/capture.py
+++ b/actfw_core/capture.py
@@ -47,7 +47,7 @@ class V4LCameraCapture(Producer[Frame[bytes]]):
         size: Tuple[int, int] = (640, 480),
         framerate: int = 30,
         expected_format: V4L2_PIX_FMT = V4L2_PIX_FMT.RGB24,
-        fallback_formats: List[V4L2_PIX_FMT] = [V4L2_PIX_FMT.YUYV, V4L2_PIX_FMT.MJPEG],
+        fallback_formats: Iterable[V4L2_PIX_FMT] = (V4L2_PIX_FMT.YUYV, V4L2_PIX_FMT.MJPEG),
         format_selector: FormatSelector = FormatSelector.DEFAULT,
     ) -> None:
         """
@@ -102,7 +102,7 @@ class V4LCameraCapture(Producer[Frame[bytes]]):
                 return 1
 
         config = None
-        fmts = [expected_format] + fallback_formats
+        fmts = [expected_format] + [f for f in fallback_formats]
         for fmt in fmts:
             expected_framerate = 1 if format_selector == V4LCameraCapture.FormatSelector.MAXIMUM else framerate
             candidates = self.video.lookup_config(width, height, expected_framerate, fmt, expected_format)
```

Reply:
I see. What a lang.

I think we should not introduce a nullable parameter as much as possible. Iterable one looks good like https://github.com/Idein/actfw-core/pull/27/files#diff-9386c1c823c4280edf402f57499277225436d14542bd8f63a158413f030a9222L13-R18.

Reply:
OK efef9b5